### ISSUE TITLE
Pod install Error in Rubygems(2.7.7)

### ISSUE THREAD

**Issue by kagenZhao** [CocoaPods/CocoaPods#7766]
Cocoapods has an error when rubygems version is 2.7.7. But there is no problem in 2.7.6.

I get an error when I use` workspace 'xxx.xcworkspace'` and `project 'xxx/xxx'`

This is My Podfile:
```
workspace 'MyExistWorkspace.xcworkspace'
target :MyProject do
    project 'MyExistedFolder/MyProjectFile'
    pod 'AFNetworking'
end
```
This is Error:
```
ArgumentError - Malformed version number string
/usr/local/lib/ruby/site_ruby/2.5.0/rubygems/version.rb:209:in `initialize'
/usr/local/lib/ruby/site_ruby/2.5.0/rubygems/version.rb:200:in `new'
/usr/local/lib/ruby/site_ruby/2.5.0/rubygems/version.rb:200:in `new'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb:123:in `embedded_content_settings'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb:68:in `generate'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/generator/xcconfig/aggregate_xcconfig.rb:39:in `save_as'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb:107:in `update_changed_file'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb:96:in `block in create_xcconfig_file'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb:93:in `each'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb:93:in `create_xcconfig_file'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb:18:in `block in install!'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/user_interface.rb:142:in `message'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/installer/xcode/pods_project_generator/aggregate_target_installer.rb:14:in `install!'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/installer/xcode/pods_project_generator.rb:181:in `block (2 levels) in install_libraries'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/installer/xcode/pods_project_generator.rb:179:in `each'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/installer/xcode/pods_project_generator.rb:179:in `block in install_libraries'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/user_interface.rb:142:in `message'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/installer/xcode/pods_project_generator.rb:167:in `install_libraries'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/installer/xcode/pods_project_generator.rb:65:in `generate!'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/installer.rb:184:in `block in generate_pods_project'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/user_interface.rb:64:in `section'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/installer.rb:183:in `generate_pods_project'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/installer.rb:119:in `install!'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/command/install.rb:41:in `run'
/usr/local/lib/ruby/gems/2.5.0/gems/claide-1.0.2/lib/claide/command.rb:334:in `run'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/lib/cocoapods/command.rb:52:in `run'
/usr/local/lib/ruby/gems/2.5.0/gems/cocoapods-1.5.2/bin/pod:55:in `<top (required)>'
/usr/local/bin/pod:23:in `load'
/usr/local/bin/pod:23:in `<main>'
```

**Comment by zedixns** [comment]
Hi @kagenZhao 

If you haven't already seen, there is an open issue already with this topic and similar issue.

Can you please move your findings there, so we can have everything in one place. We can find the solution faster that way.

The thread is #7765

**Comment by dnkoutso** [comment]
Dup of #7765

**Comment by segiddins** [comment]
This should now be fixed in CocoaPods 1.5.3!
